packages/graphql/src/contracts.js

```diff
@@ -336,71 +359,46 @@ export function setNetwork(net, customConfig) {
       decimals: '18',
       supply: '1000000000'
     })
+
+    if (shouldUseMobileBridge()) {
+      context.ognExec = new context.web3Exec.eth.Contract(
+        OriginTokenContract.abi,
+        context.ogn._address
+      )
+    }
   }
 
   context.tokens.forEach(token => {
     const contractDef =
       token.type === 'OriginToken' ? OriginTokenContract : TokenContract
-    const contract = new web3.eth.Contract(contractDef.abi, token.id)
+    let contract
+    if (shouldUseMobileBridge()) {
+      contract = new context.web3Exec.eth.Contract(contractDef.abi, token.id)
+    } else {
+      contract = new web3.eth.Contract(contractDef.abi, token.id)
+    }
+    //contract is used for calls (queries) whereas contractExec is used for transactions
     token.contract = contract
```

Comment:
I know you did not change this. But still I thought I would point this out:
I don't understand why in case mobile bridge is use we use context.web3Exec rather than context.web as value for token.contract. I thought context.web3Exec was for supposed to be used only for transactions and not for queries...